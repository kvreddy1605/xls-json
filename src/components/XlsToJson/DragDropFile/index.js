import React from 'react';
import './index.scss';

class DragDropFile extends React.Component {
	constructor(props){
		super(props);
		this.onDrop = this.onDrop.bind(this);
		this.suppress = this.suppress.bind(this);
	}

	onDrop(evt){ 
		evt.stopPropagation(); 
		evt.preventDefault();
		const files = evt.dataTransfer.files;
		if(files && files[0]) this.props.handleFile(files[0]);
	};

	suppress(evt){ 
		evt.stopPropagation(); 
		evt.preventDefault(); 
	};

	render() { 
		return (
			<div className="dragDropFile-container" onDrop={this.onDrop} onDragEnter={this.suppress} onDragOver={this.suppress}>
				{this.props.children}
			</div>
		);  
	}
};

export default DragDropFile;