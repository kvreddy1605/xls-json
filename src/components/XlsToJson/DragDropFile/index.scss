.dragDropFile-container{
    width: 400px;
    height: 200px;
    border: 2px solid #888;
    border-radius: 5px;
    padding: 10px;
}