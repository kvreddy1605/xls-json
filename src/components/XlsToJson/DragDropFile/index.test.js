import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DragDropFile from './index';

describe('DragDropFile Component Test', () => {
    it('SnapShot Test', () => {
        const wrapper = shallow(<DragDropFile />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('drop Event trigger, handleFile is called', () => {
        const handleFile = jest.fn();
        const wrapper = shallow(<DragDropFile handleFile={handleFile}/>)
        wrapper.find('div').simulate('drop', {dataTransfer: {files: [1,2,3]}, stopPropagation(){}, preventDefault(){} });
        expect(handleFile).toHaveBeenCalledWith(1);
    })
})
