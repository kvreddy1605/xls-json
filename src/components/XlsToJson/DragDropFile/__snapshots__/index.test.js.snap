// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragDropFile Component Test SnapShot Test 1`] = `
<div
  className="dragDropFile-container"
  onDragEnter={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
/>
`;
